src/cutadapt/config.vsh.yaml

```diff
@@ -0,0 +1,446 @@
+          alternatives: [-Q]
+          type: string
+          description: |
+            Quality-trimming cutoff for R2. Default: same as for R1
+        - name: --quality_base
+          type: integer
+          description: |
+            Assume that quality values in FASTQ are encoded as
+            ascii(quality + N). This needs to be set to 64 for some
+            old Illumina FASTQ files.
+          default: 33
+        - name: --poly_a
+          type: boolean_true
+          description: Trim poly-A tails
+        - name: --length
+          alternatives: [-l]
+          type: integer
+          description: |
+            Shorten reads to LENGTH. Positive values remove bases at
+            the end while negative ones remove bases at the beginning.
+            This and the following modifications are applied after
+            adapter trimming.
+        - name: --trim_n
+          type: boolean_true
+          description: Trim N's on ends of reads.
+        - name: --length_tag
+          type: string
+          description: |
+            Search for TAG followed by a decimal number in the
+            description field of the read. Replace the decimal number
+            with the correct length of the trimmed read. For example,
+            use --length-tag 'length=' to correct fields like
+            'length=123'.
+          example: "length="
+        - name: --strip_suffix
+          type: string
+          description: |
+            Remove this suffix from read names if present. Can be
+            given multiple times.
+        - name: --prefix
+          alternatives: [-x]
+          type: string
+          description: |
+            Add this prefix to read names. Use {name} to insert the
+            name of the matching adapter.
+        - name: --suffix
+          alternatives: [-y]
+          type: string
+          description: |
+            Add this suffix to read names; can also include {name}
+        - name: --rename
+          type: string
+          description: |
+            Rename reads using TEMPLATE containing variables such as
+            {id}, {adapter_name} etc. (see documentation)
+        - name: --zero_cap
+          alternatives: [-z]
+          type: boolean_true
+          description: Change negative quality values to zero.
+
+    ####################################################################
+    - name: Filtering of processed reads
+      description: |
+        Filters are applied after above read modifications. Paired-end reads are
+        always discarded pairwise (see also --pair_filter).
+      arguments:
+        - name: --minimum_length
+          alternatives: [-m]
+          type: string
+          description: |
+            Discard reads shorter than LEN.
+          default: "0"
+        - name: --maximum_length
+          alternatives: [-M]
+          type: string
+          description: |
+            Discard reads longer than LEN. Default: no limit
```

Comment:
There is no information to know that the notation `LEN:LEN2` can be used. I suppose the LEN2 is used for paired-end data. An alternative to using `type: string` could be `type: integer, multiple: true, multiple_sep: ":"` or `--minimum_length` and `--minimum_length_r2`.

Reply:
In light of some of the other remarks I would propose to use the last alternative: `--minimum_length_r2`.

Reply:
Unfortunately, the `:`-syntax covers one situation we can not handle with two distinct arguments:

> Also, one of the values can be omitted to impose no restrictions. For example, with `-m 17:`, the length of R1 must be at least 17, but the length of R2 is ignored.

This situation can also not be covered by a `multiplicatie: true` approach.

@rcannood Back to simply providing `string` which can be any of the situations cutadapt can handle?